spec.md

```diff
@@ -1420,6 +1426,113 @@ Errors](#service-broker-errors).
 }
 ```
 
+## Fetching a Service Instance
+
+If `"instances_retrievable" :true` is declared for a service in the [Catalog](#catalog-management) endpoint, Service Brokers MUST support this endpoint for all plans of the service.
+
+### Request
+
+##### Route
+`GET /v2/service_instances/:instance_id`
+
+`:instance_id` is the identifier of a previously provisioned instance.
+
+##### cURL
+```
+$ curl 'http://username:password@broker-url/v2/service_instances/:instance_id' -X GET -H "X-Broker-API-Version: 2.13"
+```
+
+### Response
+
+| Status Code | Description |
+| --- | --- |
+| 200 OK | The expected response body is below. |
+| 404 Not Found | MUST be returned if the Service Instance does not exist or if a provisioning operation is still in progress. The expected response body is `{}`. |
+
+Responses with any other status code will be interpreted as a failure.
+Brokers can include a user-facing message in the `description` field; for
+details see [Broker Errors](#service-broker-errors).
+
+##### Body
+
+For success responses, a Service Broker MUST return the following fields. For error
+responses, see [Broker Errors](#service-broker-errors).
+
+| Response field | Type | Description |
+| --- | --- | --- |
+| service_id | string | The ID of the service from the catalog that is associated with the Service Instance. |
+| plan_id | string | The ID of the plan from the catalog that is associated with the Service Instance. |
+| dashboard_url | string | The URL of a web-based management user interface for the Service Instance; we refer to this as a service dashboard. The URL MUST contain enough information for the dashboard to identify the resource being accessed (`9189kdfsk0vfnku` in the example below). Note: a Service Broker that wishes to return `dashboard_url` for a Service Instance MUST return it with the initial response to the provision request, even if the service is provisioned asynchronously. |
+| parameters | JSON object | Configuration parameters for the Service Instance. If Service Brokers cannot fetch all configuration parameters, they MUST NOT return any. |
```

Comment:
```
If Service Brokers cannot fetch all configuration parameters, they MUST NOT return any.
```

I can't remember the logic behind adding this restriction but it seems overly restrictive now. Any thoughts anyone?

Reply:
I think it was because there's no way to know what "absence" means... not there are all? Didn't change? something else?

Reply:
So if we assume a broker returns an error if it "can't fetch" parameters for some reason, then what about if a broker only saves some parameters for whatever reason?

Whilst that wouldn't lead to the nicest UI, being able to show some information about how it was configured seems better than none.